OrbitCore/CaptureData.cpp

```diff
@@ -45,12 +45,34 @@ const FunctionInfo* CaptureData::GetSelectedFunction(uint64_t function_address)
   return &selected_functions_it->second;
 }
 
+const LinuxAddressInfo* CaptureData::GetAddressInfo(uint64_t absolute_address) const {
+  auto address_info_it = address_infos_.find(absolute_address);
+  if (address_info_it == address_infos_.end()) {
+    return nullptr;
+  }
+  return &address_info_it->second;
+}
+
+void CaptureData::InsertAddressInfo(LinuxAddressInfo address_info) {
+  const uint64_t absolute_address = address_info.absolute_address();
+  const uint64_t absolute_function_address = absolute_address - address_info.offset_in_function();
+  // Ensure we know the symbols also for the resolved function address;
+  if (!address_infos_.contains(absolute_function_address)) {
+    LinuxAddressInfo function_info;
+    function_info.CopyFrom(address_info);
+    function_info.set_absolute_address(absolute_function_address);
+    function_info.set_offset_in_function(0);
+    address_infos_.emplace(absolute_function_address, function_info);
+  }
+  address_infos_.emplace(absolute_address, std::move(address_info));
+}
+
 const std::string CaptureData::kUnknownFunctionOrModuleName{"???"};
 
 const std::string& CaptureData::GetFunctionNameByAddress(uint64_t absolute_address) const {
   const FunctionInfo* function = process_->GetFunctionFromAddress(absolute_address, false);
   if (function != nullptr) {
-    return function->name();
+    return function->pretty_name();
```

Comment:
Maybe it makes sense to use FunctionUtils::GetDisplayName() function here? (it returns `pretty_name` if not empty and `name` otherwise)

Reply:
Thanks done.